src/metabase/driver/mongo/query_processor.clj

```diff
@@ -158,7 +161,26 @@
 
   DateTimeField
   (->rvalue [this]
-    (str \$ (->lvalue this)))
+    ;; omf: date/time fields using native Mongo operators
+    (let [unit (:unit this)
+      field (:field this)]
```

Comment:
don't forget you can destructure things to e.g. `[{:keys [unit field]}]` instead of `[this]`